### ISSUE TITLE
Temperature values dropping to zero

### ISSUE THREAD

**Issue by pdrsn** [thorrak/fermentrack#325]
I have done three different raspberry pi's with several ESP's on each at different locations. 
two of these fermentrack solutions experience this error where temperature from the ESP suddenly drops down to 0 and cooling stops.. I have soldered every connection on the ESP at one of the locations after experiencing this issue, but it did not help.
Any idea how I can resolve this issue, or anything I can do to help investigate this further?
The two fermentrack solutions where I experience this issue, are useless at the moment as they can't cool.
Latest fermentrack master and brewpi script v0.5 used

![45745346_348553285706210_760393736071938048_n](https://user-images.githubusercontent.com/36821179/48251613-8f4c2600-e402-11e8-8bd4-836091c96f4f.jpg)
![45400917_886983185022955_9050584935601012736_n](https://user-images.githubusercontent.com/36821179/48251615-92dfad00-e402-11e8-9ec9-2ea7a276977e.jpg)

**Comment by cpompermaier** [comment]
I have this issue very often too! I use one XLR connector with the temperature probe. My thought was that it might be a "bad connection" since the signal is low voltage. I haven't checkd the resistor that I am using with the sensor DS18B20, but it might be a too large value which makes the current too small??? Not sure if this is the culprit. But If I disconnect and reconnect the XLR connector it start logging again.

My workaround for now is to use a Python script that checks the temperature every 5 min and send an e-mail to me in case it differs too much from the set point. This helped me to be alert of possible problems, the down side is that I am not always close home to reconnect the connector...

**Comment by uSlackr** [comment]
The graph snip by the OP caught my eye because all of my graphs look that way.  Only for me this is with gravity as sensed by my Tilt.  it regularly spikes gravity high or low.  I was starting to wonder if the smoothing function isn't working.  I've been editing the data in the graph file to clean it up. 

Perhaps there is a larger issue here.

**Comment by thorrak** [comment]
> The graph snip by the OP caught my eye because all of my graphs look that way. Only for me this is with gravity as sensed by my Tilt. it regularly spikes gravity high or low. I was starting to wonder if the smoothing function isn't working. I've been editing the data in the graph file to clean it up.
> 
> Perhaps there is a larger issue here.

Which Tilt do you have (if you know)? Version 1, 2, or 3?

V1 is the “LightBlueBean” (has a picture of a fist on the chip)
V2 has “uBlox” written on the chip
V3 has “Rigado” written on the chip

Reason I ask - recent versions of the Tilt added functionality that leverages dummy temp and gravity readings to provide information on battery levels and firmware versions, which weren’t until recently being properly handled by the Tilt integration. This was fixed in a recent push to dev and is now fixed to master.

**Comment by thorrak** [comment]
> I have this issue very often too! I use one XLR connector with the temperature probe. My thought was that it might be a "bad connection" since the signal is low voltage. I haven't checkd the resistor that I am using with the sensor DS18B20, but it might be a too large value which makes the current too small??? Not sure if this is the culprit. But If I disconnect and reconnect the XLR connector it start logging again.

What microcontroller/shield are you using? @pdrsn mentioned using an ESP8266 - are you using the same or an arduino?

The issue here sounds like it's the DS18b20 dropping out on the actual controller, rather than an issue with Fermentrack specifically.

**Comment by uSlackr** [comment]
> Which Tilt do you have (if you know)? Version 1, 2, or 3?
> V1 is the “LightBlueBean” (has a picture of a fist on the chip)
> V2 has “uBlox” written on the chip
> V3 has “Rigado” written on the chip
> Reason I ask - recent versions of the Tilt added functionality that leverages dummy temp and gravity readings to provide information on battery levels and firmware versions, which weren’t until recently being properly handled by the Tilt integration (Issue #386). This was fixed in a recent push to dev which was then pushed to master last night.

I likely have a v1 unit but will check.  And I'll watch my next fermentation! Sorry for stealing the thread.

**Comment by thorrak** [comment]
> > Which Tilt do you have (if you know)? Version 1, 2, or 3?
> > V1 is the “LightBlueBean” (has a picture of a fist on the chip)
> > V2 has “uBlox” written on the chip
> > V3 has “Rigado” written on the chip
> > Reason I ask - recent versions of the Tilt added functionality that leverages dummy temp and gravity readings to provide information on battery levels and firmware versions, which weren’t until recently being properly handled by the Tilt integration (Issue #386). This was fixed in a recent push to dev which was then pushed to master last night.
> 
> I likely have a v1 unit but will check. And I'll watch my next fermentation! Sorry for stealing the thread.

No worries! Worst case post a picture of it here and I can tell you what version it is.

**Comment by pdrsn** [comment]
Switching from 3,3v to 5v on the sensors helped my problem. Its still there, but not as often as before.

**Comment by thorrak** [comment]
> Switching from 3,3v to 5v on the sensors helped my problem. Its still there, but not as often as before.

I’ve got another version of the boards I’m going to release in the next three weeks that does this (and adds decoupling caps, and reduces the size of the resistor, and...) to hopefully improve things overall. At the moment I’m waiting on some proof of concept boards to arrive before I recommend them to anyone else. 

If that doesn’t solve it, I can try to see if there’s a code fix that could be implemented to improve things but I don’t know what else could be added that the firmware doesn’t already do. I _think_ it already is attempting to reconnect disconnected sensors, but could be mistaken. Worth looking though!

**Comment by cpompermaier** [comment]
> > I have this issue very often too! I use one XLR connector with the temperature probe. My thought was that it might be a "bad connection" since the signal is low voltage. I haven't checkd the resistor that I am using with the sensor DS18B20, but it might be a too large value which makes the current too small??? Not sure if this is the culprit. But If I disconnect and reconnect the XLR connector it start logging again.
> 
> What microcontroller/shield are you using? @pdrsn mentioned using an ESP8266 - are you using the same or an arduino?
> 
> The issue here sounds like it's the DS18b20 dropping out on the actual controller, rather than an issue with Fermentrack specifically.

Hi @thorrak 
I am using : WeMos D1 mini Pro
The last problem I had (see picture below) I restarted my Raspberry PI and nothing changed, then I restarted ESP-8266, it didn’t help either, then I restarted my router thinking it might be a network  issue, still nothing happened. Then I tried to disconnect and connect again the XLR connector and started working again! It started back immediately after reconnecting, therefore I thought the connection was the culprit. It worked fine most of the time, but during the fermentation period it happened several times again, for a shorter period tough.

This was the worst case in the last fermentation, the temperature sensor was down for more than 8h:
![Log_error_temp_probe](https://user-images.githubusercontent.com/37383511/70973897-e103bd00-20a6-11ea-905e-951745b1e759.png)

It happened more than once, but not for long time as the picture above.
![log2](https://user-images.githubusercontent.com/37383511/70974110-4f487f80-20a7-11ea-82ba-e2996f034a0e.png)

**Comment by cpompermaier** [comment]
I changed the resistor to a lower valor. Before if I remember correct was 15k Ohm and now 330 Ohm. The idea is that the voltage is 3.3V and 330 Ohm will allow a maximum current of 10 mA. I am not sure the equivalent impedance of the temperature sensor, but it should be high. Since the current is much higher than bore, but within safety limits of the board, bad contact are harder to happen. Since I changed to a lower resistance, now I haven't seen the temperature drop anymore. It is running for 7 days without a glitch.

**Comment by lbussy** [comment]
I like the idea, however, that's a lot of current for the sensor.  Those are supposed to sink a max of 4.0mA.  15K is very high.  4k7 is typical for 5v and 2k2 for 3v3 systems.

**Comment by cpompermaier** [comment]
Since I changed the resistor to 330 Ohm, which is a low value, the system is working more stable. But I still have problem with the temperature dropping to zero every fermentation cycle. Next step I changed the connector from XLR to Neutrik powerCON. It improved but not 100% still. See the picture:

![image](https://user-images.githubusercontent.com/37383511/102275864-54444700-3f26-11eb-8a01-b03548caac71.png)

What is strange is that I have three temperature sensors all connected in parallel with the same 330 Ohm resistor. The ones inside the refrigerator with a 3m cable are the signals dropping, while the sensor without cable soldered with the resistor (ambient temperature) is not failing:

![image](https://user-images.githubusercontent.com/37383511/102276158-b4d38400-3f26-11eb-9efe-187ecbf3f6d1.png)

That indicates whether I have noise issue with the cable used with the sensors or the powerCON is also not providing good enough contact. 

Any suggestion?

**Comment by thorrak** [comment]
I’m pretty sure we decided this was related to either hardware issues with power, the PCB, or the temp sensors or firmware issues in BrewPi-ESP rather than issues with Fermentrack itself. Since new PCBs have since been released (alongside new BrewPi-ESP firmware) I’m going to close this issue. If the problem reoccurs please open a new issue and I’m happy to revisit.
